thunder/tests/test_grad.py

```diff
@@ -1615,3 +1615,28 @@ def foo(a):
 
     # This should call the core implementation.
     thunder.jit(foo)(a)
+
+
+def test_grad_transform_saved_for_backward_proxy():
+    from thunder.core.proxies import Proxy
+
+    def foo(a, c):
+        return a * c
+
+    a = make_tensor((2, 2), device="cpu", dtype=torch.float32, requires_grad=True)
+    c = 2.0
+
+    dynamic_jit = thunder.jit(foo, cache="symbolic values")
+    static_jit = thunder.jit(foo)
+
+    out = dynamic_jit(a, c)
+    torch.autograd.backward(out, torch.rand_like(out), retain_graph=True)
```

Comment:
It's unnecessary to run backward. The backward trace is always generated for each forward run.